src/vast_tracker.js

```diff
@@ -532,16 +531,16 @@ export class VASTTracker extends EventEmitter {
     // Use the provided fallbackClickThroughURL as a fallback
     const clickThroughURLTemplate =
       this.clickThroughURLTemplate || fallbackClickThroughURL;
+    // clone second usage of macros, which get mutated inside resolveURLTemplates
+    const clonedMacros = JSON.parse(JSON.stringify(macros));
```

Comment:
As `macros` is 1 level plain object, what about using destruct to clone the object ?
```suggestion
    const clonedMacros = {...macros};
```
Performance wise it's better I think.

Reply:
I wanted to to use `{...macros}` but eslint complained with `Unexpected token ...` and I did not wanted to fiddle with the eslint config. I'll give it a try.

Reply:
I've upgraded eslint dependencies and used babel-eslint as parser so that the spread syntax is now supported. Also had to disable the rule `no-prototype-builtins` because some parts of the code conflict with this newly enabled rule (see https://eslint.org/docs/rules/no-prototype-builtins) and I didn't want to touch that.